pyHS100/pyHS100.py

```diff
@@ -57,6 +63,39 @@ def __init__(self, ip_address, protocol=None):
             protocol = TPLinkSmartHomeProtocol()
         self.protocol = protocol
 
+        self.schedule = schedule.Schedule(self, self._query_helper)
+        self.countdown = countdown.Countdown(self, self._query_helper)
+        self.antitheft = antitheft.AntiTheft(self, self._query_helper)
+
+    def new_schedule(self):
+        """
+        Used by Schedule instance when it gets deleted.
+
+        :return: None
+        :rtype: None
+        """
+
+        self.schedule = schedule.Schedule(self, self._query_helper)
+
+    def new_countdown(self):
+        """
+        Used by Countdown instance when it gets deleted.
+
+        :return: None
+        :rtype: None
+        """
+
+        self.countdown = countdown.Countdown(self, self._query_helper)
+
+    def new_antitheft(self):
+        """
+        Used by AntiTheft instance when it gets deleted.
+        :return: None
+        :rtype: None
+        """
+
+        self.antitheft = antitheft.AntiTheft(self, self._query_helper)
+
```

Comment:
The deletion semantic (and overriding __del__) is something not very pythonic, one usually does not see ```del``` calls in the code if that's not for a very specific use case. I'd say that it'd be better to have a regular method for deletion [1], then it's explicitly clear what is being done.

[1] If Rules are to be a separate class like discussed above, maybe just delete/remove method on those? Deleting all rules can easily be done by looping all the rules and deleting them.